python_modules/dagster/dagster/_cli/job.py

```diff
@@ -567,88 +549,18 @@ def _create_external_pipeline_run(
-    check.invariant(
-        not (mode is not None and preset is not None),
-        "You may set only one of `mode` (got {mode}) or `preset` (got {preset}).".format(
-            mode=mode, preset=preset
-        ),
-    )
 
     tags = check.opt_mapping_param(tags, "tags", key_type=str)
     check.opt_sequence_param(solid_selection, "solid_selection", of_type=str)
-
-    if preset is not None:
-        pipeline_preset = external_pipeline.get_preset(preset)
-
-        if pipeline_preset.run_config is not None:
-            check.invariant(
-                (not run_config) or (pipeline_preset.run_config == run_config),
-                "The environment set in preset '{preset}' does not agree with the environment "
-                "passed in the `run_config` argument.".format(preset=preset),
-            )
-
-            run_config = pipeline_preset.run_config
-
-        # load solid_selection from preset
-        if pipeline_preset.solid_selection is not None:
-            check.invariant(
-                solid_selection is None or solid_selection == pipeline_preset.solid_selection,
-                "The solid_selection set in preset '{preset}', {preset_subset}, does not agree with"
-                " the `solid_selection` argument: {solid_selection}".format(
-                    preset=preset,
-                    preset_subset=pipeline_preset.solid_selection,
-                    solid_selection=solid_selection,
-                ),
-            )
-            solid_selection = pipeline_preset.solid_selection
-
-        check.invariant(
-            mode is None or mode == pipeline_preset.mode,
-            "Mode {mode} does not agree with the mode set in preset '{preset}': "
-            "('{preset_mode}')".format(preset=preset, preset_mode=pipeline_preset.mode, mode=mode),
-        )
-
-        mode = pipeline_preset.mode
-
-        tags = merge_dicts(pipeline_preset.tags, tags)
-
-    if mode is not None:
-        if not external_pipeline.has_mode(mode):
-            raise DagsterInvariantViolationError(
-                (
-                    "You have attempted to execute pipeline {name} with mode {mode}. "
-                    "Available modes: {modes}"
-                ).format(
-                    name=external_pipeline.name,
-                    mode=mode,
-                    modes=external_pipeline.available_modes,
-                )
-            )
-    else:
-        if len(external_pipeline.available_modes) > 1:
-            raise DagsterInvariantViolationError(
-                (
-                    "Pipeline {name} has multiple modes (Available modes: {modes}) and you have "
-                    "attempted to execute it without specifying a mode. Set "
-                    "mode property on the PipelineRun object."
-                ).format(name=external_pipeline.name, modes=external_pipeline.available_modes)
-            )
-        mode = external_pipeline.get_default_mode_name()
-
```

Comment:
Todo: review

Reply:
Everything here looks fine - just removing code that only pertains to pipelines.